forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt

```diff
@@ -0,0 +1,11 @@
+Name:Mishra, Excavation Prodigy
+ManaCost:2 R
+Types:Legendary Creature Human Artificer
+PT:2/1
+K:Haste
+A:AB$ Draw | Cost$ 1 T Discard<1/Card> | SpellDescription$ Draw a card.
+T:Mode$ DiscardedAll | ValidPlayer$ You | ValidCard$ Card.Artifact| TriggerZones$ Battlefield | ActivationLimit$ 1 | Execute$ TrigMana | TriggerDescription$ Whenever you discard one or more artifact cards, add {R}{R}. This ability triggers only once each turn.
+SVar:TrigMana:DB$ Mana | Produced$ R R
+DeckHints:Type$Artifact
+DeckHas:Ability$Discard
```

Comment:
There is a way to get the AI to prefer discarding artifacts when this is in play

Reply:
ah AILogic$ DiscardLands right? does this should work with the urza one as well right ?

Reply:
Oh nvm it's SVar:AIPreference:DiscardCost$Card.Artifact right

Reply:
Right, like Fleshgrafter